pkg/varcontext/varcontext.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2018 the Service Broker Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package varcontext
+
+import (
+	"fmt"
+
+	"github.com/spf13/cast"
+)
+
+type VarContext struct {
+	ErrorCollector
+
+	context map[string]interface{}
+}
+
+func (vc *VarContext) validate(key, typeName string, validator func(interface{}) error) {
+	val, ok := vc.context[key]
+	if !ok {
+		vc.AddError(fmt.Errorf("missing value for key %q", key))
+		return
+	}
+
+	if err := validator(val); err != nil {
+		vc.AddError(fmt.Errorf("value for %q must be a %s", key, typeName))
+	}
+}
+
+// GetString gets a string from the context, storing an error if the key doesn't
+// exist or the variable couldn't be converted to a string.
+func (vc *VarContext) GetString(key string) string {
+	vc.validate(key, "string", func(val interface{}) error {
```

Comment:
It seems like this approach causes the value to be parsed twice each time. Consider using a closure to store the result from the anonymous function's parsing, and returning that value from the outer function.